packages/commons/src/events/FieldConfig.ts

```diff
@@ -373,11 +373,19 @@ const Address = BaseField.extend({
   defaultValue: AddressFieldValue.optional()
 }).describe('Address input field – a combination of location and text fields')
 
+export const DataEntry = z.object({
+  fieldId: z.string(),
+  customValue: z
+    .object({ label: TranslationConfig, value: z.string() })
+    .optional()
+})
+export type DataEntry = z.infer<typeof DataEntry>
+
 const DataField = BaseField.extend({
   type: z.literal(FieldType.DATA),
   configuration: z.object({
     subtitle: TranslationConfig.optional(),
-    data: z.array(z.object({ fieldId: z.string() }))
+    data: z.array(DataEntry)
```

Comment:
I think we could use a model here that either the `fieldId` is used (if referring to another field from the form data) OR then the `{ label: TranslationConfig, value: z.string() }` if its a fixed value.

So the config could look like this for example:

```
data: [
  {
    fieldId: 'applicant.firstname'
  },
  {
    fieldId: 'applicant.surname'
  },
  {
    fieldId: 'applicant.dob'
  },
  {
    label: {
      defaultMessage: 'ID',
      description: 'This is the label for the field',
      id: 'v2.event.tennis-club-membership.action.print.verify.id.label'
    },
    value: 'National ID | {applicant.id}'
  }
]
```

This would make it more clear that `fieldId`s refer to other fields.

Reply:
In this case the zod typing would look something like this:

```
export const DataEntry = z.union([
  z.object({
    label: TranslationConfig,
    value: z.string()
  }),
  z.object({
    fieldId: z.string()
  })
])

export type DataEntry = z.infer<typeof DataEntry>

const DataField = BaseField.extend({
  type: z.literal(FieldType.DATA),
  configuration: z.object({
    subtitle: TranslationConfig.optional(),
    data: z.array(DataEntry)
  })
}).describe('Data field for displaying read-only data')
```

What do you think, would this be better? :)